File: diretorioprincipal/questao9.go
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao9

package main

import "fmt"

type Imprimivel interface {
    Imprimir()
}

type Relatorio struct{}

func (r Relatorio) Imprimir() {
    fmt.Println("imprimindo relatorio")
}

type Contrato struct{}

func (c Contrato) Imprimir() {
    fmt.Println("Imprimindo contrato")
}

func main() {
    var imprimivel Imprimivel

    imprimivel = Relatorio{}
    imprimivel.Imprimir()

    imprimivel = Contrato{}
    imprimivel.Imprimir()
}

File: diretorioprincipal/questao9.java
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao9

public class questao9 {

    public interface Imprimivel {
        void imprimir();
    }

    public static class Relatorio implements Imprimivel {
        @Override
        public void imprimir() {
            System.out.println("imprimindo o relatorio");
        }
    }

    public static class Contrato implements Imprimivel {
        @Override
        public void imprimir() {
            System.out.println("Imprimindo o contrato");
        }
    }

    public static void main(String[] args) {
        Imprimivel relatorio = new Relatorio();
        Imprimivel contrato = new Contrato();
        
        relatorio.imprimir();
        contrato.imprimir();
    }
}
